arch/arm/core/aarch32/cortex_m/Kconfig

```diff
@@ -305,6 +305,86 @@ config TEST_EXTRA_STACKSIZE
 
 endmenu
 
+# Implement the null pointer detection using either the Data Watchpoint and
+# Trace Unit or the Debug Monitor Exception, or the Memory Protection Unit.
+config CORTEX_M_DEBUG_NULL_POINTER_EXCEPTION
+	bool
+	help
+	  Enable and use the null pointer exception option.
+	  This is a debug feature in Cortex-M, allowing for
+	  detecting null pointer dereferencing (raising a
+	  CPU fault). Supporting the feature results in an
+	  increased code footprint.
+
+choice CORTEX_M_DEBUG_NULL_POINTER_EXCEPTION_DETECTION
+	bool "Enable and use null-pointer exception"
+	default CORTEX_M_DEBUG_NULL_POINTER_EXCEPTION_DETECTION_DWT if TEST_ARM_CORTEX_M && !ARM_NONSECURE_FIRMWARE && CPU_CORTEX_M_HAS_DWT
+	default CORTEX_M_DEBUG_NULL_POINTER_EXCEPTION_DETECTION_MPU if TEST_ARM_CORTEX_M && !ARM_NONSECURE_FIRMWARE && ARM_MPU && !CPU_CORTEX_M_HAS_DWT
+	default CORTEX_M_DEBUG_NULL_POINTER_EXCEPTION_DETECTION_NONE
+	help
+	  There are 2 implementations available one based
+	  on DWT and the other based on MPU. Use this option
+	  to select one of the options. By default the
+	  feature is disabled. In the test suite the feature
+	  is enabled and the DWT-based solution is preferred.
+
+config CORTEX_M_DEBUG_NULL_POINTER_EXCEPTION_DETECTION_NONE
```

Comment:
So, the reason for the hidden option is that it can be used as the common option for the parts of the implementation that are common to both solutions (DWT or MPU), i.e. the part that affects the linker and the fault-inspection. While the rest of the implementation (MPU configuration, or DWT setup) are specific to the applied solution.
So, common implementation parts would not need to use #if NULL_POINTER_DWT || NULL_POINTER_MPU...

Reply:
With my suggestion, `CORTEX_M_DEBUG_NULL_POINTER_EXCEPTION` will still work the same way. The only differences is that the _NONE option is gone, and users have to set 2 configs instead of 1 when enabling this (first enable, then choose backend).

Reply:
>  If you prefer keeping _NONE, I would make it clear in the help text of CORTEX_M_DEBUG_NULL_POINTER_EXCEPTION that it is not the main config, and maybe move it below the choice.

Yeap, that's fine. both done, thanks.